sympy/geometry/point.py

```diff
@@ -402,18 +403,23 @@ def distance(self, p):
         Examples
         ========
 
-        >>> from sympy.geometry import Point
+        >>> from sympy.geometry import Point, Line
         >>> p1, p2 = Point(1, 1), Point(4, 5)
+        >>> l = Line((3, 1), (2, 2))
         >>> p1.distance(p2)
         5
-
+        >>> p1.distance(l)
+        sqrt(2)
         >>> from sympy.abc import x, y
         >>> p3 = Point(x, y)
         >>> p3.distance(Point(0, 0))
         sqrt(x**2 + y**2)
 
         """
-        s, p = Point._normalize_dimension(self, Point(p))
+        from sympy.geometry.line import LinearEntity
+        if isinstance(other , LinearEntity) :
+            other = other.projection(self)
```

Comment:
How about using `return other.distance(self)`?

Reply:
I was thinking about that too but i was not sure which way to choose.
So i just decided to go with this approach.
` return other.distance(self) ` will not work for objects of other class which inherit from 'LinearEntity` like `ray` and `segment` .
But `distance` is not defined in class`ray` and `segment` so i believe this may never be needed .
 I argee with you that this approach is better .

Reply:
I need to check whether the `projection` of point p lie within the segment for distance between line and segment to make sense.

Reply:
What if other is a Plane? I think a more general, future-proof approach would be to 

1. If `other` is of a type that can be handled here (Point) handle it.
2. Try `other.distance(self)`, but catch and re-throw any AttributeError or TypeError to give a more informative error message.   

This makes evaluation is symmetric and eliminates hard-coding class references like LinearEntity.

Reply:
By mistake i have deleted my previous commit .Sorry for that.